py/tests/test_processing.py

```diff
@@ -242,7 +242,10 @@ def test_validate_sampling_configuration():
         "rules": [
             {
                 "type": "trace",
-                "sampleRate": 0.7,
+                "samplingStrategy": {
+                    "type": "sampleRate",
+                    "value": 0.7
+                },
```

Comment:
Outdated customer Relays will ignore this new field, and fail to parse this config because `sampleRate` is missing. I'm afraid the existing `unsupported` logic does not catch this.